### PR TITLE
Support tracking multiple steps in the same transaction (LEAN-3643)

### PR DESCRIPTION
**Pull request by asouqi**
Tracking multiple steps on the same node will not work like the case of [wrapInList](https://github.com/Atypon-OpenSource/manuscripts-body-editor/blob/master/src/commands.ts#L926) command. I tried to call `processChangeSteps` after we reverted all the steps that will fix that issue but will not work for other cases that need to update the positions of subsequent steps and the mapping will not help us to update steps position.

I don't want to blow up the current process that handles the steps as it makes sense to revert the step and apply it to the new transaction one by one. if the user wants to update one node with multiple steps he needs to do it with multiple transactions

### CODE REVIEW

File: src/steps/trackReplaceAroundStep.ts [hunk 1]
```diff
@@ -13,9 +13,9 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import { Slice } from 'prosemirror-model'
+import { Node as PMNode, Slice } from 'prosemirror-model'
 import type { EditorState, Transaction } from 'prosemirror-state'
-import { ReplaceAroundStep, ReplaceStep } from 'prosemirror-transform'
+import { Mapping, ReplaceAroundStep, ReplaceStep } from 'prosemirror-transform'
```
Comment (mbartenev-atypon) [inline src/steps/trackReplaceAroundStep.ts:18]: Looks like this import isn't used

**Comment by mbartenev-atypon** [comment]
"...the mapping will not help us to update steps position."
- why would it not help? @asouqi

File: src/steps/trackReplaceAroundStep.ts (hunk 1 above)
Comment (asouqi) [reply src/steps/trackReplaceAroundStep.ts:18]: updated, the eslint fix will not remove unused part!

**Comment by asouqi** [comment]
> * why would it not help?

the position of ChangeStep at the point we revert a step in the newTr will not be the same when all steps get reverted. and the mapping for the single reverted step did not work for me also

**Comment by mbartenev-atypon** [comment]
It's a bit unclear to me how ReplaceAround step is different from ReplaceStep in a way that mapping against all previous steps doesn't work for it but it works for ReplaceStep. There is nothing in the docs or in our practice that would suggest that mapping doesn't work for some type of steps. Also just general logic dictates that if steps are trackable in different but subsequent transactions, then they should be trackable in a single transaction, since it's the same thing - transaction holds a doc for each step (as implemented here too) and we work with the result of the last steps application i.e. like with the doc in a normal state.

In general I would say that changes on this PR should be merged as they make perfect sense but I don't see a reason not to go further and make it possible to make ReplaceAround steps to be trackable on mass. That can be done in a different PR. 

However, if I'm missing something and until we really support that, we should output a console error in TC plugin for cases when there are multiple ReplaceAroundSteps with an explanation that it will not be tracked properly and has to be brake into separate transactions.

If you plan to merge these changes as is, please make a comment on the ticket that we will still need to implement it later.

**Comment by asouqi** [comment]
>It's a bit unclear to me how ReplaceAround step is different from ReplaceStep in a way that mapping against all previous steps doesn't work for it but it works for ReplaceStep. There is nothing in the docs or in our practice that would suggest that mapping doesn't work for some type of steps. Also just general logic dictates that if steps are trackable in different but subsequent transactions, then they should be trackable in a single transaction, since it's the same thing - transaction holds a doc for each step (as implemented here too) and we work with the result of the last steps application i.e. like with the doc in a normal state.

no it's the same for both, the issue is about sending multiple steps that affect the same node.

> we should output a console error in TC plugin for cases when there are multiple ReplaceAroundSteps with an explanation that it will not be tracked properly and has to be broken into separate transactions.

there is already a [log](https://github.com/Atypon-OpenSource/manuscripts-track-changes-plugin/blob/233afaa21d68ad5b6b13cff1270062ebfcf98a23/src/steps/trackReplaceAroundStep.ts#L60) for that issue


>If you plan to merge these changes as is, please make a comment on the ticket that we will still need to implement it later.

I added a [test case](https://github.com/Atypon-OpenSource/manuscripts-track-changes-plugin/blob/4b50627ad0d12bd7972695d5125a00eaa602c119/test/nodes/nodes.test.ts#L234-L245) for that, with a comment

**Comment by mbartenev-atypon** [comment]
"there is already a [log](https://github.com/Atypon-OpenSource/manuscripts-track-changes-plugin/blob/233afaa21d68ad5b6b13cff1270062ebfcf98a23/src/steps/trackReplaceAroundStep.ts#L60) for that issue"

- It would be nice to state specifically that we don't support multiple replace steps so nobody needs to do any investigation again if something goes wrong

**Comment by mbartenev-atypon** [comment]
"no it's the same for both, the issue is about sending multiple steps that affect the same node."
- I still think that if that's trackable in separate transactions, it can be trackable in a single transaction too, but as I said we can have a look at it later

**Pull request merged**